x-pack/plugins/security/public/authentication/login/components/login_form/login_form.tsx

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ChangeEvent, Component, FormEvent, Fragment, MouseEvent } from 'react';
+import ReactMarkdown from 'react-markdown';
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiFieldText,
+  EuiFormRow,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { HttpStart, IHttpFetchError, NotificationsStart } from 'src/core/public';
+import { parseNext } from '../../../../../common/parse_next';
+import { LoginSelector } from '../../../../../common/login_state';
+
+interface Props {
+  http: HttpStart;
+  notifications: NotificationsStart;
+  selector: LoginSelector;
+  showLoginForm: boolean;
+  infoMessage?: string;
+  loginAssistanceMessage: string;
+}
+
+interface State {
+  loadingState:
+    | { type: LoadingStateType.None }
+    | { type: LoadingStateType.Form }
+    | { type: LoadingStateType.Selector; providerName: string };
+  username: string;
+  password: string;
+  message:
+    | { type: MessageType.None }
+    | { type: MessageType.Danger | MessageType.Info; content: string };
+}
+
+enum LoadingStateType {
+  None,
+  Form,
+  Selector,
+}
+
+enum MessageType {
+  None,
+  Info,
+  Danger,
+}
+
+export class LoginForm extends Component<Props, State> {
+  public state: State = {
+    loadingState: { type: LoadingStateType.None },
+    username: '',
+    password: '',
+    message: this.props.infoMessage
```

Comment:
__note:__ to basically override info message after first login attempt. It wasn't needed previously as Login form would either replace it with error message or redirect to another location completely. With login selector behavior changed and after user tried to login with username and password they may want to try to log in with selector and if this attempt fails I assume we don't want to display anything (or error message, but it may be redundant since we have a toast notification).